backend/src/nodes/impl/upscale/convenient_upscale.py

```diff
@@ -56,23 +57,33 @@ def convenient_upscale(
             unique_alpha = np.full(rgb.shape[:-1], unique[0], np.float32)
             return np.dstack((rgb, unique_alpha))
 
-        # Transparency hack (white/black background difference alpha)
-        black, white = with_black_and_white_backgrounds(img)
-        black_up = as_target_channels(
-            upscale(as_target_channels(black, model_in_nc, True)), 3, True
-        )
-        white_up = as_target_channels(
-            upscale(as_target_channels(white, model_in_nc, True)), 3, True
-        )
-
-        # Interpolate between the alpha values to get a more defined alpha
-        alpha_candidates = 1 - (white_up - black_up)  #  type: ignore
-        alpha_min = np.min(alpha_candidates, axis=2)
-        alpha_max = np.max(alpha_candidates, axis=2)
-        alpha_mean = np.mean(alpha_candidates, axis=2)
-        alpha = alpha_max * alpha_mean + alpha_min * (1 - alpha_mean)
-
-        return np.dstack((black_up, alpha))
+        if separate_alpha:
+            # Upscale the RGB channels and alpha channel separately
+            rgb = as_target_channels(
+                upscale(as_target_channels(img[:, :, :3], model_in_nc, True)), 3, True
+            )
+            alpha = as_target_channels(
+                upscale(as_target_channels(img[:, :, 3], model_in_nc, True)), 1, True
+            )
```

Comment:
Assuming that we are using an RGB model, we can do better. `as_target_channels` will use OpenCV's cvtColor under the hood, which basically just averages the RGB channels to get one grayscale image. This isn't ideal, as any noise will get averaged, which makes it less likely to output pure white or pure black. See #1942 for more details.

I would suggest using the same approach for denoising alpha as the BW method uses. Of course, the only applies to RGB models.